sdk/python/kfp/v2/dsl/component_bridge.py

```diff
@@ -0,0 +1,199 @@
+
+  # Check types of the reference arguments and serialize PipelineParams
+  arguments = arguments.copy()
+  for input_name, argument_value in arguments.items():
+    if isinstance(argument_value, dsl.ContainerOp):
+      raise TypeError(
+          'ContainerOp object {} was passed to component as an input argument. '
+          'Pass a single output instead.'.format(input_name))
+    elif isinstance(argument_value, dsl.PipelineParam):
+      input_type = component_spec._inputs_dict[input_name].type
+      reference_type = argument_value.param_type
+      types.verify_type_compatibility(
+          reference_type, input_type,
+          'Incompatible argument passed to the input "{}" of component "{}": '
+          .format(input_name, component_spec.name))
+
+      arguments[input_name] = str(argument_value)
+
+      if type_utils.is_artifact_type(input_type):
+        # argument_value.op_name could be none, in which case an importer node
+        # will be inserted later. Use output_artifact_key to preserve the name
+        # of pipeline parameter which is needed by importer.
+        pipeline_task_spec.inputs.artifacts[input_name].producer_task = (
+            argument_value.op_name or '')
+        pipeline_task_spec.inputs.artifacts[input_name].output_artifact_key = (
+            argument_value.name)
+      elif type_utils.is_parameter_type(input_type):
+        pipeline_task_spec.inputs.parameters[
+            input_name].runtime_value.runtime_parameter = argument_value.name
+      else:
+        raise NotImplementedError(
+            'Unsupported input type: "{}"'.format(input_type))
+    elif isinstance(argument_value, str):
+      pipeline_task_spec.inputs.parameters[
+          input_name].runtime_value.constant_value.string_value = argument_value
+    elif isinstance(argument_value, int):
+      pipeline_task_spec.inputs.parameters[
+          input_name].runtime_value.constant_value.int_value = argument_value
+    elif isinstance(argument_value, float):
+      pipeline_task_spec.inputs.parameters[
+          input_name].runtime_value.constant_value.double_value = argument_value
+    else:
+      raise NotImplementedError(
+          'Input argument supports only the following types: PipelineParam'
+          ', str, int, float. Got: "{}".'.format(argument_value))
+
+  for output in component_spec.outputs or []:
+    normalized_output_type = output.type.lower()
+    if type_utils.is_artifact_type(output.type):
+      pipeline_task_spec.outputs.artifacts[
+          output
+          .name].artifact_type.schema_title = type_utils.get_artifact_type(
+              output.type)
+    elif type_utils.is_parameter_type(output.type):
+      pipeline_task_spec.outputs.parameters[
+          output.name].type = type_utils.get_parameter_type(output.type)
+    else:
+      raise NotImplementedError('Unsupported output type: "{}"'.format(
+          output.type))
+
+  def _input_artifact_placeholder(input_key: str) -> str:
```

Comment:
Nit: `_input_artifact_path_placeholder`

Reply:
Actually, I'm changing back to use the `inputs.artifacts[].uri` placeholder given `.path` is not ready yet. Also in the future, we might support both -- recall the `use-local-path` annotation.

Reply:
The only non-working aspect of `.path` is that it resolves to the same strings as `.uri` now.